### PR TITLE
ST: Apply Roles when CO use namespace-rbac and cluster wide options properly

### PR DESCRIPTION
**Pull request by Frawless**
Signed-off-by: Jakub Stejskal <xstejs24@gmail.com>

### Type of change

- Enhancement / new feature

### Description

This PR fixes the way how roles are created for namespace-rbac mode of CO.

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Review by see-quick — APPROVED** [review]

File: systemtest/src/main/java/io/strimzi/systemtest/resources/kubernetes/RoleResource.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    public Role get(String namespace, String name) {
+        return ResourceManager.kubeClient().namespace(namespace).getRole(name);
+    }
+    @Override
+    public void create(Role resource) {
+        ResourceManager.kubeClient().namespace(resource.getMetadata().getNamespace()).createOrReplaceRole(resource);
+    }
+    @Override
+    public void delete(Role resource) {
+        ResourceManager.kubeClient().namespace(resource.getMetadata().getNamespace()).deleteRole(resource.getMetadata().getName());
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/kubernetes/RoleResource.java:34]: I think better would be use deleteRole with signature `deleteRole(namespaceName, resourceName)` inside KubeClient. I know in my refactor that these types of methods without specifying namespace explicitly end up with race-condition...

Comment (Frawless) [reply systemtest/src/main/java/io/strimzi/systemtest/resources/kubernetes/RoleResource.java]: Done

File: systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java [hunk 2]
```diff
@@ -597,16 +597,14 @@ void setupEnvironment(ExtensionContext extensionContext) throws Exception {
...
-        install.applyClusterOperatorInstallFiles(SECOND_NAMESPACE);
 
         cluster.setNamespace(FIRST_NAMESPACE);
 
         install = new SetupClusterOperator.SetupClusterOperatorBuilder()
             .withExtensionContext(extensionContext)
             .withNamespace(FIRST_NAMESPACE)
             .withWatchingNamespaces(FIRST_NAMESPACE + "," + SECOND_NAMESPACE)
-            .withBindingsNamespaces(Collections.singletonList(FIRST_NAMESPACE))
+            .withBindingsNamespaces(Arrays.asList(FIRST_NAMESPACE, SECOND_NAMESPACE))
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java:607]: `List.of(...)`?

**Review by scholzj — APPROVED** [review]
If you want to merge this, I'm fine with it. However, I wonder a bit if it is wise. If the cluster is not clean and needs the `--replace` ... can you trust it to execute the tests? It could mean a lot of things ... e.g. another Strimzi operator installed somewhere which would interfere with the tests etc.

**Comment by Frawless** [comment]
@scholzj I think we can even remove it now. The issue was caused by creating ClusterRoleBindings by applying yaml files instead of letting helm do it. I will remove it and also working your comment. Then tests again just for sure.

**Pull request merged**